lab_3_generate_by_ngrams/main.py

```diff
@@ -208,6 +322,21 @@ def build(self) -> int:
         In case of corrupt input arguments or methods used return None,
         1 is returned
         """
+        if not (isinstance(self._encoded_corpus, tuple) and self._encoded_corpus):
+            return 1
+
+        ngrams = self._extract_n_grams(self._encoded_corpus)
+
+        if not ngrams:
+            return 1
+
+        for ngram in set(ngrams):
+            if not isinstance(ngram, tuple) or not (all(isinstance(num, int) for num in ngram)):
+                return 1
+
+            self._n_gram_frequencies[ngram] = ngrams.count(ngram) / len([occ for occ in ngrams if occ[:-1] == ngram[:-1]])
```

Comment:
Good! But it would be nice to have a more understandable name for the variable `occ`